great_expectations/validator/validator.py

```diff
@@ -360,6 +377,304 @@ def inst_expectation(*args, **kwargs):
 
         return inst_expectation
 
+    def _build_expectation_configuration(
```

Comment:
As discussed, it would be great for code organization to split the self-building expectation functionality out into a separate class in a new module and invoke it here. The Validator already has many responsibilities. Choices include a new module / class in the expectation module or the rule_based_profiler module. I'm actually leaning towards the former but happy to discuss.
In addition, breaking this method up into separate methods describing what is happening would be helpful for understanding, e.g. (pseudocode, return values, type hints, etc omitted):
```
def _build_expectation_configuration(...args...):
    _validate_input_params(...args...)
    if auto:
        if user_submitted_profiler_config is not None:
            if not _check_user_submitted_profiler_config_is_complete(...args...):
                _merge_user_profiler_config_with_default(...args...)
        expectation_configuration = profiler.run(...args...)
    else:
        expectation_configuration = ExpectationConfiguration(...args...)

    return expectation_configuration

```

Reply:
@anthonyburdi How about we get this PR into the mergeable state and tackle this one as a maintenance-style enhancement shortly after?  I have other ideas for (could be same) maintenance-level PR, once this PR and PR #4269 
 are merged.  Thanks!